app/src/main/java/com/dkanada/gramophone/service/MultiPlayer.java

```diff
@@ -134,21 +133,19 @@ public MultiPlayer(Context context) {
 
         httpClient = new OkHttpClient.Builder().dispatcher(dispatcher).build();
 
-        if (exoPlayer != null) exoPlayer.release();
```

Comment:
Both this and the `simpleCache` should be always null when first executing the constructor, so I removed those lines and made the attributes final.